server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RuleActivator.java

```diff
@@ -63,17 +64,19 @@
   private final RuleIndex ruleIndex;
   private final ActiveRuleIndexer activeRuleIndexer;
   private final ActivityService activityService;
+  private final UserSession userSession;
 
   public RuleActivator(System2 system2, DbClient db, RuleIndex ruleIndex,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00107)